tests/variable/variable.cpp

```diff
@@ -386,6 +386,154 @@ TEST(Variable, getNullParentComponent)
+    EXPECT_FALSE(v1->hasDirectEquivalentVariable(v3));
+    EXPECT_FALSE(v1->hasDirectEquivalentVariable(v4));
+    EXPECT_FALSE(v1->hasDirectEquivalentVariable(v5));
+
+    EXPECT_FALSE(v2->hasDirectEquivalentVariable(v1));
+    EXPECT_FALSE(v2->hasDirectEquivalentVariable(v2));
+    EXPECT_FALSE(v2->hasDirectEquivalentVariable(v3));
+    EXPECT_FALSE(v2->hasDirectEquivalentVariable(v4));
+    EXPECT_FALSE(v2->hasDirectEquivalentVariable(v5));
+
+    EXPECT_FALSE(v3->hasDirectEquivalentVariable(v1));
+    EXPECT_FALSE(v3->hasDirectEquivalentVariable(v2));
+    EXPECT_FALSE(v3->hasDirectEquivalentVariable(v3));
+    EXPECT_FALSE(v3->hasDirectEquivalentVariable(v4));
+    EXPECT_FALSE(v3->hasDirectEquivalentVariable(v5));
+
+    EXPECT_FALSE(v4->hasDirectEquivalentVariable(v1));
+    EXPECT_FALSE(v4->hasDirectEquivalentVariable(v2));
+    EXPECT_FALSE(v4->hasDirectEquivalentVariable(v3));
+    EXPECT_FALSE(v4->hasDirectEquivalentVariable(v4));
+    EXPECT_FALSE(v4->hasDirectEquivalentVariable(v5));
+
+    EXPECT_FALSE(v5->hasDirectEquivalentVariable(v1));
+    EXPECT_FALSE(v5->hasDirectEquivalentVariable(v2));
+    EXPECT_FALSE(v5->hasDirectEquivalentVariable(v3));
+    EXPECT_FALSE(v5->hasDirectEquivalentVariable(v4));
+    EXPECT_FALSE(v5->hasDirectEquivalentVariable(v5));
+
+    libcellml::Variable::addEquivalence(v1, v2);
+    libcellml::Variable::addEquivalence(v2, v3);
+    libcellml::Variable::addEquivalence(v3, v4);
+    libcellml::Variable::addEquivalence(v4, v5);
+
+    EXPECT_FALSE(v1->hasDirectEquivalentVariable(v1));
+    EXPECT_TRUE(v1->hasDirectEquivalentVariable(v2));
+    EXPECT_FALSE(v1->hasDirectEquivalentVariable(v3));
+    EXPECT_FALSE(v1->hasDirectEquivalentVariable(v4));
+    EXPECT_FALSE(v1->hasDirectEquivalentVariable(v5));
+
+    EXPECT_TRUE(v2->hasDirectEquivalentVariable(v1));
+    EXPECT_FALSE(v2->hasDirectEquivalentVariable(v2));
+    EXPECT_TRUE(v2->hasDirectEquivalentVariable(v3));
+    EXPECT_FALSE(v2->hasDirectEquivalentVariable(v4));
+    EXPECT_FALSE(v2->hasDirectEquivalentVariable(v5));
+
+    EXPECT_FALSE(v3->hasDirectEquivalentVariable(v1));
+    EXPECT_TRUE(v3->hasDirectEquivalentVariable(v2));
+    EXPECT_FALSE(v3->hasDirectEquivalentVariable(v3));
+    EXPECT_TRUE(v3->hasDirectEquivalentVariable(v4));
+    EXPECT_FALSE(v3->hasDirectEquivalentVariable(v5));
+
+    EXPECT_FALSE(v4->hasDirectEquivalentVariable(v1));
+    EXPECT_FALSE(v4->hasDirectEquivalentVariable(v2));
+    EXPECT_TRUE(v4->hasDirectEquivalentVariable(v3));
+    EXPECT_FALSE(v4->hasDirectEquivalentVariable(v4));
+    EXPECT_TRUE(v4->hasDirectEquivalentVariable(v5));
+
+    EXPECT_FALSE(v5->hasDirectEquivalentVariable(v1));
+    EXPECT_FALSE(v5->hasDirectEquivalentVariable(v2));
+    EXPECT_FALSE(v5->hasDirectEquivalentVariable(v3));
+    EXPECT_TRUE(v5->hasDirectEquivalentVariable(v4));
+    EXPECT_FALSE(v5->hasDirectEquivalentVariable(v5));
+}
+
+TEST(Variable, hasEquivalentVariable)
+{
```

Comment:
Same as direct equivalent test too verbose.

Reply:
Agreed, 3 variables are good enough. Addressed in 941c794.